### PR TITLE
tests: remove more sleeps from ctr.bats

### PR DESCRIPTION
**Pull request by chavafg**
Change wait_until_exit function to now also
receive the expected exit status, this way we
can wait for containers that we know will fail.
Which is the case of test #3 of ctr.bats.

Also change sleeps use wait_until_exit on test #2
and #3.

Signed-off-by: Salvador Fuentes <salvador.fuentes@intel.com>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: test/ctr.bats [hunk 1]
```diff
@@ -54,12 +55,12 @@ function teardown() {
...
 	echo "$output"
 	[ "$status" -eq 0 ]
-	run sleep 5
+	wait_until_exit "$ctr_id" "1"
+	[ "$status" -eq 0 ]
 	run crictl inspect --output yaml "$ctr_id"
 	echo "$output"
 	[ "$status" -eq 0 ]
 	[[ "$output" =~ "reason: Error" ]]
-
```
Comment (wking) [inline test/ctr.bats]: This looks orthogonal.  Maybe keep this empty line?

File: test/helpers.bash [hunk 2]
```diff
@@ -502,6 +502,7 @@ function temp_sandbox_conf() {
 
 function wait_until_exit() {
 	ctr_id=$1
+	expected_exit_status=$2
```
Comment (wking) [inline test/helpers.bash]: You could simplify your diff here with a fallback during [parameter expansion][1]:

```bash
expected_exit_status="${2:-0}"
```

which would allow you to drop your current `"0"` additions and only set `"1"` when you needed it.

An alternative would be to pass this in as an environment variable instead of a positional argument, which would allow:

```bash
EXPECTED_EXIT_STATUS=1 wait_until_exit "$ctr_id"
```

and then in the function:

```bash
[[ "$output" =~ "Exit Code: ${EXPECTED_EXIT_STATUS:-0}" ]]
```

Personally, I prefer the more-explicit environment-variable approach, but I'm not a maintainer.

[1]: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02

File: test/ctr.bats (hunk 1 above)
Comment (chavafg) [reply test/ctr.bats]: agree

File: test/helpers.bash (hunk 2 above)
Comment (chavafg) [reply test/helpers.bash]: I liked the idea of the environment variable and made the modification, thanks for the advice.

**Comment by mrunalp** [comment]
@chavafg Could you port this to release-1.11 branch as well?

**Comment by chavafg** [comment]
@mrunalp sure, I'll portback to 1.11 an 1.10
